tb_plugin/torch_tb_profiler/profiler/module_op.py

```diff
@@ -0,0 +1,220 @@
+from copy import copy
+
+from .node import ModuleNode, ProfilerStepNode
+from .trace import EventTypes
+
+
+class Module:
+    def __init__(self, name, module_id):
+        self.name = name
+        self.module_id = module_id
+        self.children = []
+
+    def __hash__(self):
+        if self.children:
+            return hash((self.name, self.module_id, tuple(self.children)))
+        else:
+            return hash((self.name, self.module_id))
+
+    def __eq__(self, o) -> bool:
+        if not isinstance(o, Module):
+            return False
+
+        return (self.name == o.name and
+                self.module_id == o.module_id and
+                self.children == o.children)
+
+class ModuleStats:
+    def __init__(self, name, module_id):
+        self.name = name
+        self.module_id = module_id
+        self.occurences = 0
+        self.operators = 0
+        self.host_duration = 0
+        self.device_duration = 0
+        self.self_host_duration = 0
+        self.self_device_duration = 0
```

Comment:
oh wow didn't know that we can get the corresponding device durations as well! 😮